### ISSUE TITLE
5G of repeated warning logged:   com.hazelcast.spi.impl.AbstractInvocationFuture.warnIfSuspiciousDoubleCompletion(AbstractInvocationFuture.java:1208)

### ISSUE THREAD

**Issue by Danny-Hazelcast** [hazelcast/hazelcast#18062]
this exact WARNING message was repeated for 5Gigs in the clients logs,   printing the warning just once would suffice 

```

WARNING: hz.client_1 [HZ] [4.2-SNAPSHOT] Future.complete(Object) on completed future. Request: ClientMessage{connection=null, length=88, operation=Map.Set, isRetryable=false, correlationId=27603811, messa
geType=10f00, isEvent=false, isfragmented=false}, current value: ExceptionalResult{cause=java.lang.IllegalStateException: Cluster is in PASSIVE state! Operation: com.hazelcast.map.impl.operation.SetOperat
ion{serviceName='hz:impl:mapService', identityHash=1498203592, partitionId=178, replicaIndex=0, callId=0, invocationTime=-1 (1969-12-31 23:59:59.999), waitTimeout=-1, callTimeout=60000, tenantControl=com.
hazelcast.spi.impl.tenantcontrol.NoopTenantControl@0, name=mapBak1}}, offered value: ClientMessage{connection=ClientConnection{alive=true, connectionId=231, channel=NioChannel{/10.0.0.183:33879->/10.0.0.2
0:5701}, remoteAddress=[10.0.0.20]:5701, lastReadTime=2021-01-13 14:25:18.699, lastWriteTime=2021-01-13 14:25:18.699, closedTime=never, connected server version=4.2-SNAPSHOT}, length=19, operation=null, i
sRetryable=false, correlationId=27614360, messageType=10f01, isEvent=false, isfragmented=false}
java.lang.Exception
	at com.hazelcast.spi.impl.AbstractInvocationFuture.warnIfSuspiciousDoubleCompletion(AbstractInvocationFuture.java:1208)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.complete0(AbstractInvocationFuture.java:1185)
	at com.hazelcast.spi.impl.AbstractInvocationFuture.complete(AbstractInvocationFuture.java:1174)
	at com.hazelcast.client.impl.spi.impl.ClientInvocation.complete(ClientInvocation.java:242)
	at com.hazelcast.spi.impl.operationservice.impl.BaseInvocation.completeWithPendingResponse(BaseInvocation.java:153)
	at com.hazelcast.spi.impl.operationservice.impl.BaseInvocation.detectAndHandleBackupTimeout(BaseInvocation.java:142)
	at com.hazelcast.client.impl.spi.impl.ClientInvocationServiceImpl$CleanResourcesTask.run(ClientInvocationServiceImpl.java:305)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java:305)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:829)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76)
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102)

```

**Comment by Danny-Hazelcast** [comment]
/home/ec2-user/workspace/hot-shutdown/4.2-SNAPSHOT/2021_01_13-13_56_00/shutdown
http://jenkins.hazelcast.com/view/hot-restart/job/hot-shutdown/74/console
sh hz-bench-run hz/hot/shutdown 11

**Comment by Danny-Hazelcast** [comment]
output/HZ/HzClient6HZ/out.txt

**Comment by Holmistr** [comment]
@Danny-Hazelcast Is this a regression in 4.2 or was it present before?

**Comment by Danny-Hazelcast** [comment]
most certainly a probable regression

**Comment by sancar** [comment]
If I am not mistaken this should be a regression when moving from 3.12 and 4.0.

**Comment by sancar** [comment]
I have found the root cause. I am working on the fix. It is introduced in 4.0. It is a rare condition to happen. Events should happen in a particular(unlucky) order for this to happen so I think it could explain why the tests did not catch this before.

**Comment by Danny-Hazelcast** [comment]
@sancar  should be back a port the fix to 4.1.2. ?
